realm/realm-library/src/main/cpp/util.cpp

```diff
@@ -172,8 +172,11 @@ void ConvertException(JNIEnv* env, const char* file, int line)
     catch (std::logic_error& e) {
         ThrowException(env, IllegalState, e.what());
     }
-    catch (std::runtime_error& e) {
+    catch(realm::query_parser::InvalidQueryError& e){
```

Comment:
Are we sure that this doesn't change exception thrown on syntactic errors for existing queries?

Reply:
Great eyes :) We were waiting for the resolution on how some errors got propagated from core.

I will review whats the situation now.

Reply:
InvalidQueryError will map to IllegalStateException
InvalidQueryArgError (added in realm/realm-core#4469) will map to IllegalArgumentException

That should do, isn't it?